northstar-runtime/src/runtime/fork/init/mod.rs

```diff
@@ -348,6 +353,28 @@ impl Init {
             .map(drop)
             .expect("failed to set PR_SET_NO_NEW_PRIVS")
     }
+
+    fn enter_netns(&self) {
+        if let Some(netns) = &self.netns {
+            #[cfg(target_os = "android")]
+            let path = Path::new("/run/netns").join(netns);
+            #[cfg(not(target_os = "android"))]
+            let path = Path::new("/var/run/netns").join(netns);
+
+            if path.exists() {
+                let handle = std::fs::OpenOptions::new()
+                    .read(true)
+                    .write(false)
+                    .open(&path)
+                    .expect("failed to open netns");
+                debug!("Attaching to network namespace \"{}\"", netns);
+                setns(handle.as_raw_fd(), CloneFlags::CLONE_NEWNET).expect("failed to enter netns");
+            } else {
+                // TODO promote to hard error
```

Comment:
Can we return an error and handle it or panic during init?

Reply:
No we can't. Any error in init leads to an exit of the init process and the container is formally exited. Think we should revisit this with #372.